compiler/vm.nim

```diff
@@ -1364,6 +1364,42 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
         asgnRef(regs[ra], dest)
       else:
         globalError(c.debug[pc], "cannot evaluate cast")
+    of opcCastInt:
+      let rb = instr.regB
+      inc pc
+      let desttyp = c.types[c.code[pc].regBx - wordExcess]
+      inc pc
+      let srctyp = c.types[c.code[pc].regBx - wordExcess]
+
+      proc opIntCast(dest: var TFullReg, src: TFullReg, desttyp, srctyp: PType) =
+        if dest.kind != rkInt:
+          reset(dest); dest.kind = rkInt
+        let val = src.intVal
+        case srctyp.kind
+        of tyUInt8, tyChar:
+          dest.intVal = if desttyp.kind in {tyUInt8, tyChar}: val
```

Comment:
Types are static in Nim, the VM can eliminate casts from tyChar to tyChar, no need to check this at runtime in the VM.